/* Copyright 2019 The Waymo Open Dataset Authors. All Rights Reserved.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
==============================================================================*/

syntax = "proto2";

package waymo.open_dataset;

import "waymo_open_dataset/dataset.proto";
import "waymo_open_dataset/protos/metrics.proto";

// If your inference results are too large to fit in one proto, you can shard
// them to multiple files by sharding the inference_results field.
message Submission {
  // These values correspond to the tasks on the waymo.com/open site.
  enum Task {
    UNKNOWN = 0;
    DETECTION_2D = 1;
    DETECTION_3D = 2;
    TRACKING_2D = 3;
    TRACKING_3D = 4;
  }

  // This specifies which task this submission is for.
  optional Task task = 1;

  // This must be set as the full email used to register at waymo.com/open.
  optional string account_name = 13;
  // This name needs to be short, descriptive and unique. Only the latest result
  // of the method from a user will show up on the leaderboard.
  optional string unique_method_name = 2;
  repeated string authors = 3;
  optional string affiliation = 4;
  optional string description = 5;

  // Link to paper or other link that describes the method.
  optional string method_link = 6;

  // Sensors used.
  repeated LaserName.Name lidars = 7;
  repeated CameraName.Name cameras = 8;

  // Number of frames used.
  optional int32 number_past_frames_exclude_current = 9;
  optional int32 number_future_frames_exclude_current = 10;

  // Inference results.
  optional Objects inference_results = 11;
}
